### PR TITLE
Enable Post-Authorisation Pay Run Edits (and Fix Invoice Update Across Multiple Currencies Issue)

### PR DESCRIPTION
**Pull request by Publicker**
This PR introduces the capability to edit a Pay run after it has been sent for authorisation, enhancing flexibility in the Pay run management process. Additionally, addresses a bug affecting the update of invoices within a Pay run when enabling/disabling invoices of different currencies. Prior to this fix, only invoices of just one currency were being saved correctly.

# Changes
- **Pay Run Edit Flow**: Added functionality to allow for the editing of a Pay run even after it has been sent for authorisation. It's important to note that this edit functionality, along with the associated 'Edit Pay Run' button, will be disabled once the Pay run has been approved. This is to maintain the integrity of the Pay run and associated payouts, which are created at the approval stage.
- **Update Invoices of Different Currencies**: Resolved an issue where updating invoices of different currencies within the same Pay run resulted in only one currency type being saved. Now, invoices of all specified currencies are correctly updated and saved.

# Demo Video

https://github.com/nofrixion/nofrixion.business/assets/52673485/18b84ec7-481e-4a80-95ae-a523d0e07b0d

### CODE REVIEW

**Comment by sipsorcery** [comment]
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/215.

**Review by donalnofrixion — APPROVED** [review]

**Comment by donalnofrixion** [comment]
🔥🔥

File: packages/components/src/components/ui/pages/PayrunDetails/PayrunDetails.tsx [hunk 1]
```diff
@@ -156,14 +160,19 @@ const PayrunDetails: React.FC<PayrunDetailsProps> = ({
...
+          ...invoicesChanged.reduce((acc, invoice) => {
+            if (!acc) return {}
+            acc[invoice.id] = invoice.isEnabled ?? false
+            return acc
+          }, {} as PayrunUpdate['invoices']),
+        }
       }
     })
 
+    console.log('changesMade.invoices', changesMade.invoices)
```
Comment (sipsorcery) [inline packages/components/src/components/ui/pages/PayrunDetails/PayrunDetails.tsx:174]: Debug?

**Review by sipsorcery — APPROVED** [review]

File: packages/components/src/components/ui/pages/PayrunDetails/PayrunDetails.tsx (hunk 1 above)
Comment (Publicker) [reply packages/components/src/components/ui/pages/PayrunDetails/PayrunDetails.tsx:174]: Oops, thanks.

**Review by sauravmaiti22 — APPROVED** [review]

**Review by pabloi-nf — APPROVED** [review]

**Pull request merged**